psa-crypto/src/types/algorithm.rs

```diff
@@ -755,6 +769,199 @@ impl From<AsymmetricEncryption> for psa_crypto_sys::psa_algorithm_t {
     }
 }
 
+#[cfg(feature = "interface")]
+impl TryFrom<psa_crypto_sys::psa_algorithm_t> for Mac {
+    type Error = Error;
+    fn try_from(alg: psa_crypto_sys::psa_algorithm_t) -> Result<Self> {
+        if psa_crypto_sys::PSA_ALG_IS_FULL_LENGTH_MAC(alg) {
+            Ok(Mac::FullLength(alg.try_into()?))
+        } else {
```

Comment:
Bit confused what you mean here. If it ends up in this method to begin with, it _has_ to be a MAC algorithm, otherwise the code above this method isn't working correctly. The block starting on line 544 (the `TryFrom<psa_crypto_sys::psa_algorithm_t> for Algorithm` is what directs MAC algorithms to this particular method.

Reply:
> it _has_ to be a MAC algorithm

`impl`s are "public", though - if someone has a `psa_algorithm_t` and they do `Mac::try_from(alg)`, this gets called, regardless of what that `alg` was.

Reply:
Ah, I see what you mean. There should be a fail case for `TryFrom`, otherwise you'd use `From`.